docs/06-ReusingTestCode.md

```diff
@@ -0,0 +1,636 @@
+<?php
+$I->seePageReloaded();
+$I->seeClassIsLoaded($classname);
+$I->dontSeeUserExist($user);
+?>
+```
+And then use them in your tests:
+
+```php
+<?php
+$I->seePageReloaded();
+$I->seeClassIsLoaded('FunctionalTester');
+$I->dontSeeUserExist($user);
+?>
+```
+
+You can define asserts by using assertXXX methods in modules. Not all PHPUnit assert methods are included in modules, but you can use PHPUnit static methods from the `PHPUnit_Framework_Assert` class to leverage all of them.
+
+```php
+<?php
+
+function seeClassExist($class)
+{
+    $this->assertTrue(class_exists($class));
+    // or
+    \PHPUnit_Framework_Assert::assertTrue(class_exists($class));
+}
+?>
+```
+
+In your helpers you can use these assertions:
+
+```php
+<?php
+
+function seeCanCheckEverything($thing)
+{
+    $this->assertTrue(isset($thing), "this thing is set");
+    $this->assertFalse(empty($any), "this thing is not empty");
+    $this->assertNotNull($thing, "this thing is not null");
+    $this->assertContains("world", $thing, "this thing contains 'world'");
+    $this->assertNotContains("bye", $thing, "this thing doesn`t contain 'bye'");
+    $this->assertEquals("hello world", $thing, "this thing is 'Hello world'!");
+    // ...
+}
+?>
+```
+
+### Resolving Collisions
+
+What happens if you have two modules that contain the same named actions?
+Codeception allows you to override actions by changing the module order.
+The action from the second module will be loaded and the action from the first one will be ignored.
+The order of the modules can be defined in the suite config.
+
+However, some of modules may conflict with each others. In order to avoid confusion which module is used in first place, Framework modules, PhpBrowser, and WebDriver can't be used together. `_conflicts` method of module is used to specify which class or interface it conflicts to. Codeception will throw exception if there will be a module enabled which matches the provided criteria.
+
+### Connecting Modules
+
+It's possible that you will need to access internal data or functions from other modules. For example, for your module you might need access responses or internal actions of modules.
+
+Modules can interact with each other through the `getModule` method. Please note that this method will throw an exception if the required module was not loaded.
+
+Let's imagine that we are writing a module that reconnects to a database. It's supposed to use the dbh connection value from the Db module.
+
+```php
+<?php
+
+function reconnectToDatabase() {
+    $dbh = $this->getModule('Db')->dbh;
+    $dbh->close();
+    $dbh->open();
+}
+?>
+```
+
+By using the `getModule` function, you get access to all of the public methods and properties of the requested module. The dbh property was defined as public specifically to be available to other modules.
+
+If you want to extend functionality of standard module you may connect to it and by using public properties and methods your custom action or assertion:
```

Comment:
extend functionality of **a** standard module

and methods **create** your custom action or assertion